### PR TITLE
Adapt `NetworkPolicy`s

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security networking
/kind enhancement
/platform aws

**What this PR does / why we need it**:
This PR adapts the `NetworkPolicy`s as following:

- The `oidc-webhook-authenticator` `Service` is now annotated with `networking.resources.gardener.cloud/from-all-webhook-targets-allowed-ports` to allow it being reached from the `kube-apiserver` of shoots (see https://github.com/gardener/gardener/pull/7907). The webhook mutating the pod labels of the `kube-apiserver` is deprecated and can be cleaned up in a future release.
- The custom pod label selectors for Prometheus is adapted similar to https://github.com/gardener/gardener/pull/7907/commits/b14706780cb74902404aa44351bd39124aee48ae.

**Which issue(s) this PR fixes**:
Related to https://github.com/gardener/gardener/issues/7352

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: charts/gardener-extension-shoot-oidc-service/templates/service.yaml [hunk 1]
```diff
@@ -9,14 +9,16 @@ metadata:
...
-    networking.resources.gardener.cloud/from-policy-pod-label-selector: all-seed-scrape-targets
+    networking.resources.gardener.cloud/from-all-seed-scrape-targets-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
     networking.resources.gardener.cloud/from-policy-allowed-ports: '[{"port":{{ .Values.metrics.port }},"protocol":"TCP"}]'
     networking.resources.gardener.cloud/namespace-selectors: '[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}}]'
     networking.resources.gardener.cloud/pod-label-selector-namespace-alias: extensions
 {{- if .Values.ignoreResources }}
     resources.gardener.cloud/ignore: "true"
 {{- end }}
+    # TODO: This label approach is deprecated and no longer needed in the future. Remove them as soon as gardener/gardener@v1.75 has been released.
+    networking.resources.gardener.cloud/from-world-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
```
Comment (dimityrmirchev) [inline charts/gardener-extension-shoot-oidc-service/templates/service.yaml:20]: Maybe I am missing something, but isn't this label still needed (the server port)? I do not see it removed in other PRs. Maybe you meant to change the labels regarding the `metrics` port? See [this](https://github.com/gardener/gardener/commit/b14706780cb74902404aa44351bd39124aee48ae#diff-0f942ac1144b309bca6a3b86b204f6a3e3daa1877a9651e1c3d291f121b72927L8-L9) as an example.

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Thanks! I will probably test this tomorrow, but one question after a quick first look.

File: charts/gardener-extension-shoot-oidc-service/templates/service.yaml [hunk 2]
```diff
@@ -10,13 +10,15 @@ metadata:
   namespace: {{ .Release.Namespace }}
   annotations:
     networking.resources.gardener.cloud/from-world-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
-    networking.resources.gardener.cloud/from-policy-pod-label-selector: all-seed-scrape-targets
-    networking.resources.gardener.cloud/from-policy-allowed-ports: '[{"port":{{ .Values.metrics.port }},"protocol":"TCP"}]'
+    networking.resources.gardener.cloud/from-all-seed-scrape-targets-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
```
Comment (dimityrmirchev) [inline charts/gardener-extension-shoot-oidc-service/templates/service.yaml:13]:
```suggestion
    networking.resources.gardener.cloud/from-all-seed-scrape-targets-to-ports: '[{"protocol":"TCP","port":{{ .Values.metrics.port }}}]'
```

I guess this should also be changed

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Thanks! I think we need one more change. The other looks good

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
When do you plan the next release containing this PR?

**Comment by dimityrmirchev** [comment]
I will release it later today

**Pull request merged**